python/run-tests.py

```diff
@@ -161,8 +161,13 @@ def run_individual_python_test(target_dir, test_name, pyspark_python):
 
 def get_default_python_executables():
     python_execs = [x for x in ["python2.7", "python3.6", "pypy"] if which(x)]
-    if "python2.7" not in python_execs:
-        LOGGER.warning("Not testing against `python2.7` because it could not be found; falling"
+    if ("python3.6" not in python_execs) and which("python3"):
+        LOGGER.warning("Not testing against `python3.6` because it could not be found; falling"
+                       " back to `python3` instead")
+        python_execs.insert(0, "python3")
```

Comment:
It's not neccessary. This script already provides an option to specify python executables. `--python-executables`

Reply:
Yes there's an `--python-executables`. But maybe we can improve the default case.
Suppose one newbie user install py3.5 and user run `python/run-tests`. But the master test script will automatically test on py2.7, he will be confusing.

Reply:
In that case, we can just remove python 2.7. I think that's not what this JIRA targets

Reply:
I revert the change here.